vision/cloud-client/src/main/java/com/example/vision/Detect.java

```diff
@@ -0,0 +1,405 @@
+
+    BatchAnnotateImagesResponse response = visionApi.batchAnnotateImages(requests);
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+
+    for (AnnotateImageResponse res : responses) {      
+      if (res.hasError()) out.printf("Error: %s\n", res.getError().getMessage());
+
+      for (EntityAnnotation annotation : res.getLogoAnnotationsList()) {
+        out.println(annotation.getDescription());
+      }
+    }
+  }
+  
+  /**
+   * Detects text in the specified image.
+   * @param filePath The path to the file to detect text in.
+   * @param out A {@link PrintStream} to write the detected text to.
+   * @throws IOException 
+   */
+  static public void detectText(String filePath, PrintStream out) throws IOException {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+
+    FileInputStream file = new FileInputStream(filePath);
+    ByteString imgBytes = ByteString.copyFrom(toByteArray(file));
+    
+    Image img = Image.newBuilder().setContent(imgBytes).build();    
+    Feature feat = Feature.newBuilder().setType(Type.TEXT_DETECTION).build();
+    AnnotateImageRequest request = AnnotateImageRequest.newBuilder()
+        .addFeatures(feat)
+        .setImage(img)
+        .build();
+    requests.add(request);
+
+    BatchAnnotateImagesResponse response = visionApi.batchAnnotateImages(requests);
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+
+    for (AnnotateImageResponse res : responses) {
+      if (res.hasError()) out.printf("Error: %s\n", res.getError().getMessage());
+
+      for (EntityAnnotation annotation : res.getTextAnnotationsList()) {
+        out.printf("Text: %s\n", annotation.getDescription());
+      }
+    }
+  }
+  
+  /**
+   * Detects image properties such as color frequency from the specified image.
+   * @param filePath The path to the file to detect properties.
+   * @param out A {@link PrintStream} to write 
+   * @throws IOException 
+   */
+  static public void detectProperties(String filePath, PrintStream out) throws IOException {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+
+    FileInputStream file = new FileInputStream(filePath);
+    ByteString imgBytes = ByteString.copyFrom(toByteArray(file));
+    
+    Image img = Image.newBuilder().setContent(imgBytes).build();    
+    Feature feat = Feature.newBuilder().setType(Type.IMAGE_PROPERTIES).build();
+    AnnotateImageRequest request = AnnotateImageRequest.newBuilder()
+        .addFeatures(feat)
+        .setImage(img)
+        .build();
+    requests.add(request);
+
+    BatchAnnotateImagesResponse response = visionApi.batchAnnotateImages(requests);
+    List<AnnotateImageResponse> responses = response.getResponsesList();
+
+    for (AnnotateImageResponse res : responses) {      
+      if (res.hasError()) out.printf("Error: %s\n", res.getError().getMessage());
+
+      DominantColorsAnnotation colors = res.getImagePropertiesAnnotation().getDominantColors();
```

Comment:
A comment alluding to the presence of other possible annotations, with a shortlink to the doc (http://g.co/cloud/... could come in handy there) might be helpful here.